From c683f14ef654ac62480d6f5bce9609106e09d5e2 Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Tue, 3 Apr 2012 17:29:09 +0000
Subject: [PATCH] [CODEC-96] Base64 encode() method is no longer thread-safe,
 breaking clients using it as a shared BinaryEncoder. Make Context ivar final.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/codec/trunk@1309062 13f79535-47bb-0310-9956-ffa450edef68
---
 .../org/apache/commons/codec/binary/BaseNCodecInputStream.java  | 2 +-
 .../org/apache/commons/codec/binary/BaseNCodecOutputStream.java | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/org/apache/commons/codec/binary/BaseNCodecInputStream.java b/src/main/java/org/apache/commons/codec/binary/BaseNCodecInputStream.java
index 3fea132ff6..9c8ba60dd6 100644
--- a/src/main/java/org/apache/commons/codec/binary/BaseNCodecInputStream.java
+++ b/src/main/java/org/apache/commons/codec/binary/BaseNCodecInputStream.java
@@ -38,7 +38,7 @@ public class BaseNCodecInputStream extends FilterInputStream {
 
     private final byte[] singleByte = new byte[1];
 
-    private Context context = new Context();
+    private final Context context = new Context();
 
     protected BaseNCodecInputStream(InputStream in, BaseNCodec baseNCodec, boolean doEncode) {
         super(in);
diff --git a/src/main/java/org/apache/commons/codec/binary/BaseNCodecOutputStream.java b/src/main/java/org/apache/commons/codec/binary/BaseNCodecOutputStream.java
index 885e7f0687..3068107dc4 100644
--- a/src/main/java/org/apache/commons/codec/binary/BaseNCodecOutputStream.java
+++ b/src/main/java/org/apache/commons/codec/binary/BaseNCodecOutputStream.java
@@ -38,7 +38,7 @@ public class BaseNCodecOutputStream extends FilterOutputStream {
 
     private final byte[] singleByte = new byte[1];
 
-    private Context context = new Context();
+    private final Context context = new Context();
 
     // TODO should this be protected?
     public BaseNCodecOutputStream(OutputStream out, BaseNCodec basedCodec, boolean doEncode) {
